--- XLD/XLDPluginManager.m.orig
+++ XLD/XLDPluginManager.m
@@ -53,13 +53,13 @@
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5
 				NSArray *archArray = [bundle executableArchitectures];
 #if defined(__x86_64__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitectureX86_64)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitectureX86_64)]] == NSNotFound) continue;
 #elif defined(__i386__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitectureI386)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitectureI386)]] == NSNotFound) continue;
 #elif defined(__ppc__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitecturePPC)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitecturePPC)]] == NSNotFound) continue;
 #elif defined(__aarch64__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitectureARM64)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitectureARM64)]] == NSNotFound) continue;
 #endif
 #endif
 				if(![[bundle infoDictionary] objectForKey:@"NSPrincipalClass"]) continue;
@@ -80,13 +80,13 @@
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_5
 				NSArray *archArray = [bundle executableArchitectures];
 #if defined(__x86_64__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitectureX86_64)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitectureX86_64)]] == NSNotFound) continue;
 #elif defined(__i386__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitectureI386)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitectureI386)]] == NSNotFound) continue;
 #elif defined(__ppc__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitecturePPC)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitecturePPC)]] == NSNotFound) continue;
 #elif defined(__aarch64__)
-				if([archArray indexOfObject:@(NSBundleExecutableArchitectureARM64)] == NSNotFound) continue;
+				if([archArray indexOfObject:[NSNumber numberWithInt:((int)NSBundleExecutableArchitectureARM64)]] == NSNotFound) continue;
 #endif
 #endif
 				if(![[bundle infoDictionary] objectForKey:@"NSPrincipalClass"]) continue;

--- XLD/XLDController.m.orig
+++ XLD/XLDController.m
@@ -2828,7 +2828,7 @@
 		[task setTrack:track];
 	}
 	if(removeFlag && [o_warnBeforeConversion state] == NSOnState) {
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__ppc__)
 		__block int ret;
 		if (![NSThread isMainThread]) {
 			dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);

--- XLD/main.m.orig
+++ XLD/main.m
@@ -45,7 +45,7 @@
 			Method toMethod   = class_getInstanceMethod([NSBundle class],@selector(mod_infoDictionary2));
 			method_exchangeImplementations(fromMethod, toMethod);
 			XLDDarkModeSupportEnabled = 1;
-			[pref registerDefaults:@{@"NSLayerPerformanceUpdates": @(YES)}];
+			[NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:@"NSLayerPerformanceUpdates"];
 		}
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
 		else {
